specs/core/0_beacon-chain.md

```diff
@@ -863,6 +896,19 @@ Perform the following checks:
 
 For each validator index `v` in `intersection`, if `state.validators[v].status` does not equal `PENALIZED`, then run `exit_validator(v, state, penalize=True, current_slot=block.slot)`
 
+#### PROPOSER_SLASHING
+
+```python
+{
+    'proposer_index': 'uint24',
+    'proposal1_data': ProposalSignedData,
+    'proposal1_sig': '[uint256]',
+    'proposal2_data': ProposalSignedData,
+    'proposal1_sig': '[uint256]',
+}
+```
+For each `proposal_sig`, verify that `BLSVerify(pubkey=validators[proposer_index].pubkey, msg=proposal_data, sig=proposal_sig)` passes. Verify that `proposal1_data.slot == proposal2_data.slot` but `proposal1 != proposal2`. If `state.validators[proposer_index].status` does not equal `PENALIZED`, then run `exit_validator(proposer_index, state, penalize=True, current_slot=block.slot)`
```

Comment:
Naming: unify `proposal_sig` and `proposer_signature`?

Reply:
Changed sig -> signature, but I think "proposal" should stay as it's talking about two signatures of two different proposals, but from the same proposer.